packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -283,27 +283,31 @@ const Select = <
 
   // Normalize the options array to accommodate the various
   // data types that might be passed
-  const normalizedOptions = useMemo(
-    () =>
-      options.map((option) => {
-        if (typeof option === "object") {
-          /**
-           * If the value of `option?.value is an empty string, we need to make sure that we
-           * set an empty string to `value` in the normalized option so that the select component
-           * can potentially set it as the selected one in the text input
-           */
-          const value =
-            option?.value === "" ? option.value : option.value || option.text;
-          return {
-            text: option.text,
-            value,
-            type: option.type === "heading" ? "heading" : "option",
-          };
-        }
-        return { text: option, value: option, type: "option" };
-      }),
-    [options],
-  );
+  const [normalizedOptions, normalizedOptionsMap] = useMemo(() => {
+    const normalizedOptions = options.map((option) => {
+      if (typeof option === "object") {
+        /**
+         * If the value of `option?.value is an empty string, we need to make sure that we
+         * set an empty string to `value` in the normalized option so that the select component
+         * can potentially set it as the selected one in the text input
+         */
+        const value =
+          option?.value === "" ? option.value : option.value || option.text;
+        return {
+          text: option.text,
+          value,
+          type: option.type === "heading" ? "heading" : "option",
+        };
+      }
+
+      return { text: option, value: option, type: "option" };
+    });
+
+    const normalizedOptionsMap = new Map(
+      normalizedOptions.map((option) => [option.value, option]),
+    );
+    return [normalizedOptions, normalizedOptionsMap];
+  }, [options]);
```

Comment:
I would make this a separate `useMemo` so it's less confusing why a `useMemo`'s return value looks like a `useState`:
```suggestion
  const normalizedOptions = useMemo(() => {
    const normalizedOptions = options.map((option) => {
      if (typeof option === "object") {
        /**
         * If the value of `option?.value is an empty string, we need to make sure that we
         * set an empty string to `value` in the normalized option so that the select component
         * can potentially set it as the selected one in the text input
         */
        const value =
          option?.value === "" ? option.value : option.value || option.text;
        return {
          text: option.text,
          value,
          type: option.type === "heading" ? "heading" : "option",
        };
      }

      return { text: option, value: option, type: "option" };
    });
  }, [options]);
  
  const normalizedOptionsMap = useMemo(() => (new Map(
    normalizedOptions.map((option) => [option.value, option]),
  ), [normalizedOptions]);
```

But I'm also wondering why we have this in the first place. I'll post another comment where I saw the reason.

Reply:
I am unsure if this really confuses people like use state... it just returns a tuple. I think putting it in another use memo makes it an extra render, no?

Reply:
Naw, two `useMemo` won't cause a re-render. Only a `setState` would do that, and that's not even true. Sometimes, it batches them together.